_posts/2023-05-17-otp-26-highlights.md

```diff
@@ -0,0 +1,892 @@
+done (passed successfully)
+```
+
+## Pull request
+
+[dialyzer: Add incremental analysis mode](https://github.com/erlang/otp/pull/5997)
+
+# argparse: A command line parser for Erlang
+
+New in OTP 26 is the
+[argparse](https://www.erlang.org/doc/man/argparse.html) module, which
+simplifies parsing of the command line in
+[escripts](https://www.erlang.org/doc/man/escript.html).  `argparse`
+was implemented by Maxim Fedorov.
+
+To show only a few of the features, let's implement the command-line
+parsing for an escript called `ehead`, inspired by the Unix command
+[head](https://en.wikipedia.org/wiki/Head_(Unix)):
+
+```erlang
+#!/usr/bin/env escript
+%% -*- erlang -*-
+
+main(Args) ->
+    argparse:run(Args, cli(), #{progname => ehead}).
+
+cli() ->
+    #{
+      arguments =>
+          [#{name => lines, type => {integer, [{min, 1}]},
+             short => $n, long => "-lines", default => 10,
+             help => "number of lines to print"},
+           #{name => files, nargs => nonempty_list, action => extend}],
+      handler => fun(Args) ->
+                         io:format("~p\n", [Args])
+                 end
+     }.
+```
+
+As currently written, the `ehead` script will simply print the
+arguments collected by `argparse` and quit.
+
+If `ehead` is run without any arguments an error message will be
+shown:
+
+```
+$ ehead
+error: ehead: required argument missing: files
+Usage:
+  ehead [-n <lines>] [--lines <lines>] <files>...
+
+Arguments:
+  files       files
+
+Optional arguments:
+  -n, --lines number of lines to print (int >= 1, 10)
+```
+
+The message tells us that at least one file name must be given:
+
+```
+$ ehead foo bar baz
+#{lines => 10,files => ["foo","bar","baz"]}
+```
+
+Since the command line was valid, `argparse` collected the arguments
+into a map, which was then printed by the `handler` fun.
+
+The number of lines to be printed from each file defaults to `10`, but
+can be changed using either the `-n` or `--lines` option:
+
+```
+$ ehead -n 42 foo bar baz
+#{lines => 42,files => ["foo","bar","baz"]}
+$ ehead foo --lines=42 bar baz
+#{lines => 42,files => ["foo","bar","baz"]}
+$ ehead --lines 42 foo bar baz
+#{lines => 42,files => ["foo","bar","baz"]}
+$ ehead foo bar --lines 42 baz
+#{lines => 42,files => ["foo","bar","baz"]}
+```
+
+Attempting to give the number of lines as `0` results in an error message:
+
+```
+$ ehead -n 0 foobar
+error: ehead: invalid argument for lines: 0 is less than accepted minimum
+Usage:
+  ehead [-n <lines>] [--lines <lines>] <files>...
+
+Arguments:
+  files       files
+
+Optional arguments:
+  -n, --lines number of lines to print (int >= 1, 10)
+```
+
+## Pull request
+
+[[argparse] Command line parser for Erlang](https://github.com/erlang/otp/pull/6852)
+
+
+# SSL: Safer defaults
+
+In OTP 25, the default client options for the `ssl` appliction would
+allow setting up a connection without verifying that the server used
+trusted CA certificates. For example:
+
```

Comment:
I suggest you write it like this instead:

In OTP 25, the default options for the `ssl` application would
allow clients to set up a connection to a server without verifying that  servers authenticity
by checking its certificate chain. For example

Reply:
I have rephrased that paragraph.